### PR TITLE
#410 created integration test DyUserITCase using a real (local) dynamo server

### PR DESCRIPTION
**Pull request by adko-pl**
@davvd Please find a reviewer

### CODE REVIEW

**Comment by davvd** [comment]
Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@longtimeago review this one, please

File: src/test/java/com/thindeck/dynamo/DyUserTestITCase.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * @author Adam Siemion (adam.siemion.null@lemonsoftware.pl)
+ * @version $Id$
+ * @todo Implement a test mthod for user repos
+ */
+public class DyUserTestITCase {
+    /**
+     * Sample user URN.
+     */
+    private static final String USER_URN = "urn:test:1";
```
Comment (longtimeago) [inline src/test/java/com/thindeck/dynamo/DyUserTestITCase.java]: no need to take this variable out from the test - it should be method local

File: src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                        Integer.parseInt(System.getProperty("dynamo.port"))
+                    )
+                )
+            ),
+            "td-"
+        );
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (longtimeago) [inline src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java]: this doc is redundant

File: src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+     */
+    @Override
+    public final AmazonDynamoDB aws() {
+        return this.region.aws();
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (longtimeago) [inline src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java]: same here

File: src/test/java/com/thindeck/dynamo/ClasspathRule.java [hunk 4]
```diff
@@ -94,6 +94,7 @@ public boolean apply(final Class<?> input) {
                         && !name.endsWith("ITCase")
                         && !name.endsWith("ClasspathRule")
                         && !name.endsWith("RepoRule")
+                        && !name.equals(RegionLocalDynamo.class.getName())
```
Comment (longtimeago) [inline src/test/java/com/thindeck/dynamo/ClasspathRule.java]: why did you add this condition?
to pass ```ImmutabilityTests``` you need to make ```RegionLocalDynamo``` immutable.
In your case it's pretty simple - just add  ```@Immutable``` to that class

**Comment by longtimeago** [comment]
@adko-pl Good job! See several remarks in the code above

**Comment by adko-pl** [comment]
@longtimeago fixed the code review comments

**Comment by longtimeago** [comment]
@adko-pl thank you!

**Comment by longtimeago** [comment]
@rultor please, merge

File: src/test/java/com/thindeck/dynamo/DyUserTestITCase.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Integration tests for {@link com.thindeck.dynamo.DyUser}
+ * using a real (local) DynamoDB server.
+ *
+ * @author Adam Siemion (adam.siemion.null@lemonsoftware.pl)
+ * @version $Id$
+ * @todo Implement a test method for user repos
```
Comment (carlosmiranda) [inline src/test/java/com/thindeck/dynamo/DyUserTestITCase.java]: I see a `@todo` comment here, but no ticket number and estimate? Should be something like, `@todo #410:30min`.

See http://www.yegor256.com/2009/03/04/pdd.html for more details.

File: src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import com.jcabi.manifests.Manifests;
+
+/**
+ * Region for tests.
+ *
+ * @author Adam Siemion (adam.siemion.null@lemonsoftware.pl)
+ * @version $Id$
+ */
+@Immutable
+public class RegionLocalDynamo implements Region {
```
Comment (carlosmiranda) [inline src/test/java/com/thindeck/dynamo/RegionLocalDynamo.java]: It would be better if we declare this class to be final instead of declaring its methods final.

File: src/test/java/com/thindeck/dynamo/DyUserTestITCase.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+ */
+public class DyUserTestITCase {
```
Comment (carlosmiranda) [inline src/test/java/com/thindeck/dynamo/DyUserTestITCase.java]: Let's declare the entire class final instead of marking each of its methods final. There's no reason why anyone would extend this class.

**Comment by carlosmiranda** [comment]
@adko-pl I have a few of my own comments above. :)

**Comment by adko-pl** [comment]
@carlosmiranda fixed your code review comments

**Comment by carlosmiranda** [comment]
@rultor merge please

**Comment by davvd** [comment]
@longtimeago Many thanks! **16 mins** were added to your account in Transaction ID `51657127`. you're getting extra minutes here (c=1). +16 to your rating, your total score is [+1864](http://www.netbout.com/b/35024?open=rating)

**Comment by davvd** [comment]
@rultor please deploy
